src/main/java/com/liferay/faces/util/cache/internal/CacheFactoryImpl.java

```diff
@@ -15,64 +15,60 @@
  */
 package com.liferay.faces.util.cache.internal;
 
+import javax.faces.context.ExternalContext;
+import javax.faces.context.FacesContext;
+
 import com.liferay.faces.util.cache.Cache;
 import com.liferay.faces.util.cache.CacheFactory;
+import com.liferay.faces.util.config.WebConfigParam;
 
 
 /**
  * @author  Kyle Stiemann
  */
 public class CacheFactoryImpl extends CacheFactory {
 
-	@Override
-	public <K, V> Cache<K, V> getCache() {
-		return new CacheImpl<K, V>();
-	}
+	// Private Final Data Members
+	private final int defaultInitialCacheCapacity;
 
-	@Override
-	public <K, V> Cache<K, V> getCache(int initialCapacity) {
+	public CacheFactoryImpl() {
 
-		validateInitialCapacity(initialCapacity);
+		FacesContext facesContext = FacesContext.getCurrentInstance();
 
-		return new CacheImpl<K, V>(initialCapacity);
-	}
+		if (facesContext != null) {
 
-	@Override
-	public <K, V> Cache<K, V> getCache(int initialCapacity, int maxCapacity) {
+			ExternalContext externalContext = facesContext.getExternalContext();
 
-		validateInitialCapacity(initialCapacity);
-		validateMaxCapacity(maxCapacity);
+			defaultInitialCacheCapacity = WebConfigParam.DefaultInitialCacheCapacity.getIntegerValue(externalContext);
+		}
 
-		return new CacheMaxCapacityLRUImpl<K, V>(initialCapacity, maxCapacity);
+		// Otherwise we are in a test environment so use the default value.
+		else {
```

Comment:
@stiemannkj1 I know this sounds like a lot of work for seemingly little benefit, but I think that mocking the FacesContext and ExternalContext would be closer to the spirit of testing than making the code aware of the test environment. As it stands now, the `getIntegerValue(externalContext)` method call only gets tested in-container.